### DISCUSSION TITLE
What happened to the UI?

### DISCUSSION THREAD

**Discussion by Imaynotbehere4long** [AssetRipper/AssetRipper#1503] [Q&A] [answered]
I haven't updated AssetRipper in a while, and the first thing I notice is that it opens a new browser tab instead of being its own window like before, which I'm not fond of (closing the tab doesn't close the command-line window and vice versa like how previous versions close both when you closed one). Worse, when I click File > Open, the file-selection window always appears BEHIND my browser window! Then, I notice I have to click a "view files" button instead of the list showing up automatically upon loading like before. Plus, "export all files of selected type" isn't under the Export drop down anymore, even if I select a file. Is that feature just gone now? That was the main reason I used this program. I managed to find "export primary content" after some trial and error, but what if I don't want ALL the primary content, just audio or 2D textures, etc.? Why were these changes made? If these features are still available, where can I find them?

While I'm at it, I do have an extra-feature suggestion: let us sort assets alphabetically by asset-type the way Unity Asset Bundle Extractor did. Have an option that lists all assets from all sub-directories in one giant list, then let us order that list by asset-type so we can scroll down until we see the exact type of asset we're looking for (audio clip, 2d texture, etc.), and--of course--let us just export those assets of that specific type again.

Or maybe you could have tabs like what currently shows up when you select an asset, and each tab corresponds to a different asset-type so that clicking said tab shows a list of all of that type of asset in every currently-open file.

**Comment by ds5678** [comment]
You seem angry at me. Why?

**Reply by Imaynotbehere4long** [reply]
Maybe not you specifically, but I admit I'm upset that a program I've used reliably for the past few years ended up becoming less user friendly by several metrics (if not outright removing various features), with the only noticeable upside being compatibility with more recent Unity versions. Now that I've answered your question, will you answer my questions? I asked them in earnest.

**Comment by ds5678** [comment, marked as answer]
Your post has a tone of entitlement. Please be mindful that you received software updates for free, that you are continuing to do so, and that many alternatives (eg AssetStudio) have been abandoned. Do not bite the hand that feeds.

> Why did the UI change happen?

It may be disappointing to hear, but many changes to AssetRipper happened simply because I wanted to, including the switch to an HTML UI. In that particular case, I had a strong dislike for Avalonia and wanted to learn more about web development.

In fact, skill development has been a primary motivation throughout my time developing AssetRipper. Big features almost always featured (haha) some form of skill development, and last year, I got a software engineering job.

> Why in the browser and not a self contained window?

I actually used Electron initially. Both the community and I ended up hating that, so we pivoted to the current style. However, there's an alternative to Electron called Photino that I want to look into.

> Why does the file picker appear behind the browser?

That's atypical, and I suspect your browser is subtly pushing itself to the front. I don't think there's anything I can do about it.

> Why remove the selective export feature?

I didn't like how it was structured. I'm not opposed to it returning in some form, especially now that primary content extraction exists.

> Why not more user friendly?

I plan to dedicate some time to quality of life improvements. Lately, they've fallen by the wayside as I make room for more pressing development, like the recent push for IL2Cpp Recovery.

**Reply by Imaynotbehere4long** [reply]
Thanks.
